rir/src/compiler/translations/rir_2_pir_compiler.h

```diff
@@ -0,0 +1,47 @@
+#ifndef RIR_2_PIR_COMPILER_H
+#define RIR_2_PIR_COMPILER_H
+
+#include "../opt/cleanup.h"
+#include "../opt/delay_env.h"
+#include "../opt/delay_instr.h"
+#include "../opt/elide_env.h"
+#include "../opt/force_dominance.h"
+#include "../opt/inline.h"
+#include "../opt/scope_resolution.h"
```

Comment:
I don't like those dependencies here. Can you move the constructor implementation to the cpp file?